base/bitarray.jl

```diff
@@ -987,25 +988,68 @@ function deleteat!(B::BitVector, inds)
 
     (p, s) = y
     checkbounds(B, p)
+    p isa Bool && throw(ArgumentError("invalid index $p of type Bool"))
```

Comment:
We can throw an error here (and not just warn about deprecation), because earlier it already threw an error as a side effect of lack of conversion of `p` to `Int` where needed later. (same below)

The current behavior is:
```
julia> deleteat!(falses(5), Any[true])
ERROR: MethodError: no method matching copy_chunks!(::Vector{UInt64}, ::Bool, ::Vector{UInt64}, ::Int64, ::Int64)

julia> deleteat!(falses(5), Any[1, true])
ERROR: ArgumentError: indices must be unique and sorted
```
Now in both cases we will throw `throw(ArgumentError("invalid index $p of type Bool"))`